src/nodes/Geometry3D/IndexedFaceSet.js

```diff
@@ -865,8 +878,6 @@ x3dom.registerNodeType(
                                     else if (hasColorInd && !colPerVert) { c2 = +colorInd[faceCnt]; }
                                     else if (colPerVert) { c2 = p2; }
                                     else { c2 = faceCnt; }
-                                    t = 3;
-
```

Comment:
With the fix, there are no case t=3 anymore, the creation of any triangle is set in case t=2